File: buysellStock.py
'''
Time Complexity: O(n)
Space Complexity:O(1)
Did this code successfully run on Leetcode : Yes
Explanation:   In this solution if the stock had increased before and now decreased, that means there was a
peak and now there is a valley so capture that diff as profit and keep adding all of them to get the answer
'''

def maxProfit(self, prices: List[int]) -> int:
    # In this solution if the stock had increased before and now decreased, that means there was a
    # peak and now there is a valley so capture that diff as profit and keep adding all of them to
    # get the answer
    if len(prices) < 2:
        return 0

    maxProfit = 0

    for i in range(1, len(prices)):
        if prices[i] > prices[i - 1]:
            maxProfit += prices[i] - prices[i - 1]

    return maxProfit

File: BuySellStock.java
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No 
class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int i = 0;
        while (i + 1 < prices.length) {
            if (prices[i + 1] > prices[i]) {
                max = max + prices[i + 1] - prices[i];
            }
            i++;
        }
        return max;
    }
}